commands/root.go

```diff
@@ -16,55 +16,131 @@
 package commands
 
 import (
+	"context"
 	"fmt"
 	"log"
 	"os"
 	"strings"
 
+	"github.com/c-bata/go-prompt"
 	"github.com/spf13/cobra"
+	"github.com/spf13/pflag"
 
 	clustercmd "github.com/hazelcast/hazelcast-commandline-client/commands/cluster"
 	mapcmd "github.com/hazelcast/hazelcast-commandline-client/commands/types/map"
 	"github.com/hazelcast/hazelcast-commandline-client/internal"
+	"github.com/hazelcast/hazelcast-commandline-client/internal/cobraprompt"
 )
 
 var (
-	cfgFile   string
-	addresses string
-	cluster   string
-	token     string
-	rootCmd   = &cobra.Command{
+	RootCmd = &cobra.Command{
 		Use:   "hzc {cluster | help | map} [--address address | --cloud-token token | --cluster-name name | --config config]",
 		Short: "Hazelcast command-line client",
 		Long:  "Hazelcast command-line client connects your command-line to a Hazelcast cluster",
 		Example: "`hzc map --name my-map put --key hello --value world` - put entry into map directly\n" +
 			"`hzc help` - print help",
 		RunE: func(cmd *cobra.Command, args []string) error {
+			if _, err := internal.MakeConfig(); err != nil {
+				return err
+			}
 			return cmd.Help()
 		},
 	}
 )
 
+func addressAndClusterNamePrefix() (prefix string, useLive bool) {
+	if config := internal.Configuration; config != nil {
+		return fmt.Sprintf("hzc %s@%s> ", config.Cluster.Network.Addresses[0], config.Cluster.Name), true
+	}
+	return "hzc address@clusterName> ", true
```

Comment:
Is it probable that the configuration is `nil`? I think when this function is called, we should have already connected to the cluster, shouldn't we?

Reply:
Thank you, fixed as suggested